pallets/msa/src/lib.rs

```diff
@@ -229,7 +292,7 @@ impl<T: Config> Pallet<T> {
 					.try_insert(index, key.clone())
 					.map_err(|_| Error::<T>::KeyLimitExceeded)?;
 
-				Ok(())
+				on_success(msa_id)
```

Comment:
why do we need to this?

Substrate does not take care of transactions correctly. When we have multiple mutates(non-nested) after each other if the first one is successful and the second one fails the first mutate changes will not be reverted. That's why we need to nest these mutates so if one of them fail then all the rest also reverted.